tests/unit/lib/utilities/filters/test_filters.py

```diff
@@ -0,0 +1,183 @@
+        self.assertEquals(in_range_value, None)
+
+    def test_open_reject(self):
+        """If OPEN_REJECT filter applied value will be discarded if: lower_bound < value < upper_bound"""
+        open_reject_filter = RangeFilter(Type.OPEN_REJECT, self.lower_bound, self.upper_bound)
+        out_range_value = open_reject_filter.filter(10)
+        in_range_value = open_reject_filter.filter(15)
+        self.assertEquals(out_range_value, 10)
+        self.assertEquals(in_range_value, None)
+
+    def test_closed_open_reject(self):
+        """If CLOSED_OPEN_REJECT filter applied value will be discarded if: lower_bound <= value < upper_bound"""
+        closed_open_reject = RangeFilter(Type.CLOSED_OPEN_REJECT, self.lower_bound, self.upper_bound)
+        out_range_value = closed_open_reject.filter(20)
+        in_range_value = closed_open_reject.filter(10)
+        self.assertEquals(out_range_value, 20)
+        self.assertEquals(in_range_value, None)
+
+    def test_open_closed_reject(self):
+        """If OPEN_CLOSED_REJECT filter applied value will be discarded if: lower_bound < value <= upper_bound"""
+        open_closed_reject = RangeFilter(Type.OPEN_CLOSED_REJECT, self.lower_bound, self.upper_bound)
+        out_range_value = open_closed_reject.filter(10)
+        in_range_value = open_closed_reject.filter(20)
+        self.assertEquals(out_range_value, 10)
+        self.assertEquals(in_range_value, None)
+
+    """
+    Filters bounded at one end
+    """
+    def test_less_than(self):
+        """If LESS_THAN filter applied value will be accepted if: value < lower_bound"""
+        less_than_filter = RangeFilter(Type.LESS_THAN, self.lower_bound, self.upper_bound)
+        out_range_value = less_than_filter.filter(10)
+        in_range_value = less_than_filter.filter(0)
+        self.assertEquals(out_range_value, None)
+        self.assertEquals(in_range_value, 0)
+
+    def test_at_most(self):
+        """If AT_MOST filter applied value will be accepted if: value <= lower_bound"""
+        at_most_filter = RangeFilter(Type.AT_MOST, self.lower_bound, self.upper_bound)
+        out_range_value = at_most_filter.filter(20)
+        in_range_value = at_most_filter.filter(0)
+        self.assertEquals(out_range_value, None)
+        self.assertEquals(in_range_value, 0)
+
+    def test_greater_than(self):
+        """If GREATER_THAN filter applied value will be accepted if: value > upper_bound"""
+        greater_than_filter = RangeFilter(Type.GREATER_THAN, self.lower_bound, self.upper_bound)
+        out_range_value = greater_than_filter.filter(20)
+        in_range_value = greater_than_filter.filter(30)
+        self.assertEquals(out_range_value, None)
+        self.assertEquals(in_range_value, 30)
+
+    def test_at_least(self):
+        """If AT_LEAST filter applied value will be accepted if: value >= upper_bound"""
+        at_least_filter = RangeFilter(Type.AT_LEAST, self.lower_bound, self.upper_bound)
```

Comment:
In general, this are a lot of test cases and if someone modifies the numbers in future, all test cases might have to be fixed. I think better idea will be to define less_than_lower_bound= 5 lower_bound=10, middle_value = 15, upper_bound = 20, more_than_upper_bound=25 etc. and then write test cases to validate for all.